Wrap install.sh in a function

tianon opened the pull request:
With tabs, instead of spaces, and a comment about why we're wrapped up in a function. :wink:

Closes #12241

ewindisch committed:
Wrap installer in a function

This will assure that the install script will not
begin executing until after it has been downloaded should
it be utilized in a 'curl | bash' workflow.

Signed-off-by: Eric Windisch <eric@windisch.us>

tianon committed:
Commonalize more bits of install.sh (especially standardizing around "cat <<-EOF")

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

thaJeztah commented on hack/install.sh:
```diff
@@ -20,213 +20,218 @@ command_exists() {
 
-EOF
-exit 1
+# wrapped up in a function so that we have some proection against only getting half the file during "curl | sh"
```
Typo s/proection/protection

thaJeztah commented on hack/install.sh:
```diff
@@ -20,213 +20,218 @@ command_exists() {
+			echo
+			echo 'Remember that you will have to log out and back in for this to take effect!'
+			echo
+			exit 0
```
(unrelated to this PR) Should this message also be wrapped in a function? The exact same message is included twice in this script (yeah, nit-picking)

tiborvass replied:
+1

tianon commented:
:+1: updated with a function for the shared message, and updated the larger uses of repeated `echo >&2` to use `cat >&2 <<-EOF` appropriately instead.

thaJeztah commented:
Thanks!

tianon commented:
Updated! :+1:

tiborvass commented:
LGTM

estesp commented:
LGTM
